x402-piece-gateway/test/test-helpers.js

```diff
@@ -0,0 +1,134 @@
+import { exact } from 'x402/schemes'
+
+export const TEST_PAYEE = '0xc83dBFDf61616778537211A7E5CA2e87ec6cF0ed'
+export const TEST_CID =
+  'baga6ea4seaqaleibb6ud4xeemuzzpsyhl6cxlsymsnfco4cdjka5uzajo2x4ipa'
+export const TEST_PRICE = '1000000' // 1 USDC (6 decimals)
+
+/** Helper to create mock useFacilitator for dependency injection */
+export function createMockUseFacilitator({ verifyResult, settleResult } = {}) {
+  return () => ({
+    verify: async () => verifyResult ?? { isValid: true, payer: TEST_PAYEE },
+    settle: async () =>
+      settleResult ?? {
+        success: true,
+        transaction: '0xabc123',
+        network: 'base-sepolia',
+      },
+  })
+}
+
+/** Helper to create a valid test payment header */
+export function createTestPaymentHeader() {
+  const payment = {
+    x402Version: 1,
+    scheme: 'exact',
+    network: 'base-sepolia',
+    payload: {
+      signature: '0x' + '00'.repeat(65),
+      authorization: {
+        from: TEST_PAYEE,
+        to: TEST_PAYEE,
+        value: TEST_PRICE,
+        validAfter: '0',
+        validBefore: String(Math.floor(Date.now() / 1000) + 3600),
+        nonce: '0x' + '00'.repeat(32),
+      },
+    },
+  }
+  return exact.evm.encodePayment(payment)
+}
+
+/**
+ * Creates a data set in the database
+ *
+ * @param {Env} env
+ * @param {Object} options
+ * @param {number} options.dataSetId
+ * @param {string} options.payerAddress
+ * @param {boolean} options.withCDN
+ */
+export async function withDataSet(
+  env,
+  { dataSetId = 0, payerAddress = TEST_PAYEE, withCDN = true } = {},
+) {
+  await env.DB.prepare(
+    `INSERT INTO data_sets (id, payer_address, with_cdn)
+     VALUES (?, ?, ?)`,
+  )
+    .bind(String(dataSetId), payerAddress.toLowerCase(), withCDN)
+    .run()
+}
+
+/**
+ * Creates a piece in the database
+ *
+ * @param {Env} env
+ * @param {Object} options
+ * @param {string} options.pieceId
+ * @param {number} options.dataSetId
+ * @param {string} options.pieceCid
+ * @param {string} [options.x402Price]
+ */
+export async function withPiece(
+  env,
+  { pieceId = 0, dataSetId = 0, pieceCid = TEST_CID, x402Price = null } = {},
+) {
+  await env.DB.prepare(
+    `INSERT INTO pieces (id, data_set_id, cid, x402_price)
+     VALUES (?, ?, ?, ?)`,
+  )
+    .bind(String(pieceId), String(dataSetId), pieceCid, x402Price)
+    .run()
+}
```

Comment:
It would be great to have a shared library of test data builders, so that we use the same helpers in all packages across the monorepo.

It's ok to perform this refactoring after this pull request is landed. Perhaps create a follow-up tech-debt issue?

Reply:
There's already an [issue](https://github.com/filbeam/worker/issues/352) opened, I'm going to reference this comment there.